packages/next/src/client/components/gracefully-degrade-error-boundary.tsx

```diff
@@ -0,0 +1,85 @@
+'use client'
+
+import { Component, createRef } from 'react'
+import type { ErrorInfo, ReactNode, RefObject } from 'react'
+
+interface ErrorBoundaryProps {
+  children: ReactNode
+}
+
+interface ErrorBoundaryState {
+  hasError: boolean
+}
+
+const errorBannerHtml = `\
+<div id="next-graceful-error" style="position: fixed; bottom: 16px; right: 16px; background: #fff; color: #000; padding: 12px 20px; z-index: 999;">
```

Comment:
Do we need to worry about the design for this as it could potentially still show in bot caches and such? I think we make the this part a little more noticeable/maybe red background. cc @pranathip for thoughts. 

![CleanShot 2025-04-08 at 14 35 10@2x](https://github.com/user-attachments/assets/d1c0022c-2d11-4524-af5d-1c1c10c03731)

Reply:
if we dont have to show the error then we can drop the banner and do even simpler with leaving the page in error state.

https://github.com/vercel/next.js/pull/77916#discussion_r2034175436